compiler/bindings/python/iree/compiler/tools/import_onnx/__main__.py

```diff
@@ -37,19 +46,213 @@
+                for out in graph_info.output_map.values()
+            ]
+            ftype = onnx_importer.FunctionType.get(input_types, output_types)
+            func_op = onnx_importer.func_dialect.FuncOp(
+                func_name,
+                ftype,
+                ip=InsertionPoint(body),
+                visibility="private" if private else None,
+            )
+            block = func_op.add_entry_block(
+                [Location.name(k) for k in graph_info.input_map.keys()]
+            )
+        imp = IREENodeImporter(
+            graph_info,
+            parent_op=func_op,
+            block=block,
+            context_cache=cc,
+            module_op=module_op,
+            module_cache=mc,
+            max_numel=max_numel,
+        )
+        for node_name, input_value in zip(graph_info.input_map.keys(), block.arguments):
+            imp._nv_map[node_name] = input_value
+        imp._populate_graph_attrs(func_op)
+        return imp
+
+    def import_initializer(
+        self, initializer: onnx.TensorProto, extern_name: Optional[str] = None
+    ) -> Value:
+        # If an explicitly specified name is given, use that; otherwise, pick
+        # up the name from the tensor proto itself
+        iname = extern_name if extern_name else initializer.name
+        dims = list(initializer.dims)
+        acc = 1
+        for d in dims:
+            acc = acc * d
+        if acc < self.max_numel:
+            imported_tensor = super().import_initializer(initializer)
+            self._nv_map[iname] = imported_tensor
+            return imported_tensor
+
+        x, t = self.create_tensor_global(initializer)
+        vtensor_type = self._cc.get_vtensor_type(
+            tuple(initializer.dims), self._cc.tensor_element_type(initializer.data_type)
+        )
+
+        with InsertionPoint(self._b), Location.name(iname):
+            old_op = util.GlobalLoadOp(t, x)
+            converted_value = Operation.create(
+                "torch_c.from_builtin_tensor",
+                results=[vtensor_type],
+                operands=[old_op.result],
+            ).result
+
+        self._nv_map[iname] = converted_value
+        tensor_as_array = numpy_helper.to_array(initializer)
+        self.param_archive.add_buffer(x, tensor_as_array)
+        return converted_value
+
+
 def main(args: argparse.Namespace):
     model_proto = load_onnx_model(args)
     context = Context()
     model_info = onnx_importer.ModelInfo(model_proto)
     m = model_info.create_module(context=context).operation
-    imp = onnx_importer.NodeImporter.define_function(model_info.main_graph, m)
+
+    imp: Any = None
+    if args.externalize_params:
+        imp = IREENodeImporter.define_function(model_info.main_graph, m, args.max_numel)
+    else:
+        if args.max_numel:
+            warnings.warn(
+                "'--max-numel' has no effect until externalization is enabled with '--externalize-params'"
+            )
```

Comment:
Since `max_numel` has a default value, this warning will always trigger. I think it is fine to remove this warning altogether.

Reply:
At least in my opinion, it might be more intuitive to call it `min_numel` since its the minimum number of elements required to externalize a param. Or maybe `numel_thresh`? Not sure what the best naming would be.

Reply:
Removed the warning already. I think we should use threshold, as max or min will both be confusing.